pkgs/games/hmcl/default.nix

```
@@ -0,0 +1,58 @@
+{ lib, stdenv
+, fetchurl
+, jre
+, makeWrapper
+, makeDesktopItem
+, copyDesktopItems
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hmcl";
+  version = "3.5.3.223";
+
+  src = ./.;
+
+  hmcl = fetchurl {
+    url = "https://github.com/huanghongxun/HMCL/releases/download/v${version}/HMCL-${version}.jar";
+    sha256 = "sha256-8g2FMvAiAKfxJUY0G7wl6d44wOpVsknFHGZ85IkOzFc=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [ makeWrapper copyDesktopItems ];
+
+  desktopItem = makeDesktopItem {
+    name = "hmcl";
+    exec = "hmcl";
+    icon = "hmcl";
+    comment = "A Minecraft Launcher which is multi-functional, cross-platform and popular";
+    desktopName = "HMCL";
+    categories = [ "Game" ];
+  };
+
+  desktopItems = [ desktopItem ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/opt/hmcl
+    install -Dm644 $hmcl $out/opt/hmcl/hmcl.jar
+    install -Dm644 ${./icon.png} $out/share/icons/hicolor/48x48/apps/hmcl.png
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    makeWrapper ${jre}/bin/java $out/bin/hmcl \
+      --argv0 "hmcl" \
+      --add-flags "-jar $out/opt/hmcl/hmcl.jar" \
+      --prefix PATH : ${lib.makeBinPath [ jre ]} \
+      --set JAVA_HOME ${lib.getBin jre}
+  '';
+
+  meta = with lib; {
+    description = "A Minecraft Launcher which is multi-functional, cross-platform and popular";
+    homepage = "https://hmcl.huangyuhui.net";
+    maintainers = with maintainers; [ candyc1oud ];
+    license = licenses.gpl3;
+  };
```

`sourceProvenance` is needed for all java-based derivations, and we might as well set `platforms`.
```suggestion
    sourceProvenance = with sourceTypes; [ binaryBytecode ];
    inherit (jre.meta) platforms;
  };
```